classes/rule.php

```diff
@@ -136,6 +136,30 @@ public function load_from_form($form) {
 
         $this->data_save();
     }
+    /**
+     * Load a rule from a trigger object.
+     * @param $trigger
+     */
+    public function load_rule_from_trigger($trigger) {
+
+        $rule = new rule;
+        $rule->id                  = $trigger->id;
+        $rule->context             = $trigger->context;
+        $rule->datatype            = $trigger->datatype;
+        $rule->name                = $trigger->name;
+        $rule->priority            = $trigger->priority;
+        $rule->parent              = $trigger->parent;
+        $rule->lengthtype          = $trigger->lengthtype;
+        $rule->lengthfromduedate   = $trigger->lengthfromduedate;
+        $rule->elapsedtype         = $trigger->elapsedtype;
+        $rule->elapsedfromrequest  = $trigger->elapsedfromrequest;
+        $rule->role                = $trigger->role;
+        $rule->action              = $trigger->action;
+        $rule->data                = $trigger->data;
+        $rule->parentrule          = $trigger->parentrule;
+        $rule->children            = $trigger->children;
```

Comment:
We're basically doing the same thing a number of times here (pretty much $rule->$key = $trigger->$key;) so we could refactor to adhere to the DRY principle a bit better.

The trigger object that we get has all of the properties that we need, but we need the object we return to be an instance of "rule", so we do need to create a rule object and then transfer the properties to that.

We can use get_object_vars to get all of the properties of $trigger as an array, then iterate through that to apply each one to our $rule. As an added precaution we can avoid adding properties that we don't want by doing something like what already happens in load_from_form and only adding properties that exist for our rule object.